File: bridge/src/bridge_leaderboards.cpp
#if defined(DM_PLATFORM_HTML5)
#include "bridge_leaderboards.h"
#include "bridge.h"

int bridge::leaderboards::type(lua_State* L) {
    return getString(L, js_bridge_leaderboards_type);
}

int bridge::leaderboards::setScore(lua_State* L) {
    return makeCallbackLeaderboardsSetScore(L, js_bridge_leaderboards_setScore, false);
}

int bridge::leaderboards::getEntries(lua_State* L) {
    return makeCallbackWithString(L, js_bridge_leaderboards_getEntries, true);
}

#endif

File: bridge/bridge_leaderboards.lua
local leaderboards = {}

function leaderboards.type()
	return "not_available"
end

function leaderboards.set_score(id, score, on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

function leaderboards.get_entries(id, on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

function leaderboards.show_native_popup(id, on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

return leaderboards